src/material-experimental/mdc-slider/slider.ts

```diff
@@ -170,6 +172,22 @@ export class MatSlider implements AfterViewInit, OnDestroy {
     }
   }
 
+  /**
+   * Sets up the initial state of the slider thumb inputs.
+   *
+   * The slider thumbs need this extra step because are passed in via `ng-content`,
+   * and therefore have no way of knowing which slider thumb they correspond to.
+   *
+   * This method needs to return a promise in order to avoid throwing an
+   * ExpressionChangedAfterItHasBeenCheckedError error.
+   */
+  _initInputs(): Promise<void> {
+    return Promise.resolve().then(() => {
+      this._inputs.get(0)?._init(this._isRange() ? Thumb.START : Thumb.END);
+      this._inputs.get(1)?._init(Thumb.END);
+    });
+  }
```

Comment:
It's somewhat strange for the parent component to initialize its children like this- why can't they initialize themselves in their own `ngOnInit`?